core/src/main/java/org/apache/brooklyn/policy/InvokeEffectorOnCollectionSensorChange.java

```diff
@@ -128,10 +132,25 @@ public void setEntity(EntityLocal entity) {
     }
 
     @Override
+    protected boolean isBusy() {
+        return super.isBusy() || moreUpdatesComing.get();
+    }
+
+    @Override
     public void onEvent(SensorEvent<Collection<?>> event) {
         final Set<Object> newValue = event.getValue() != null
                 ? new LinkedHashSet<>(event.getValue())
                 : ImmutableSet.of();
+        if (isBusySensorEnabled()) {
+            // There are more events coming that this policy hasn't been notified of if the
+            // value received in the event does not match the current value of the sensor.
+            final Collection<?> sensorVal = entity.sensors().get(getTriggerSensor());
+            final Set<Object> sensorValSet = sensorVal != null
+                ? new LinkedHashSet<>(sensorVal)
+                : ImmutableSet.of();
+            final boolean moreComing = !Objects.equals(newValue, sensorValSet);
+            moreUpdatesComing.set(moreComing);
+        }
```

Comment:
Why not do this in the parent class? It would be useful in "InvokeEffectorOnSensorChange" as well.